src/js/trove/table.js

```diff
@@ -366,11 +596,39 @@
         
         'get-column': runtime.makeMethod1(function(_, col_name) {
           ffi.checkArity(2, arguments, "get-column", true);
+          runtime.checkString(col_name);
           if(!hasColumn(col_name)) {
             ffi.throwMessageException("The table does not have a column named `"+col_name+"`.");
           }
           return runtime.ffi.makeList(getColumn(col_name));
         }),
+
+        'select-columns': runtime.makeMethod1(function(_, colnames) {
+          ffi.checkArity(2, arguments, "get-column", true);
+          runtime.checkList(colnames);
+
+          var colnamesList = ffi.toArray(colnames);
+          if(colnamesList.length === 0) {
+            throw ffi.throwMessageException("zero-columns");
```

Comment:
This is actually a well-formedness error on tables right now. But since you can't drop columns, and can't select zero columns, it can be entirely enforced syntactically.